src/NuGet.Core/NuGet.Common/PathUtil/DirectoryUtility.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace NuGet.Common
+{
+    /// <summary>
+    /// Directory operation helpers.
+    /// </summary>
+    public static class DirectoryUtility
+    {
+        private static object s_sharedFolderCreationLock = new object();
+
+        /// <summary>
+        /// Creates all directories and subdirectories in the specified path unless they already exist.
+        /// New directories can be read and written by all users.
+        /// </summary>
+        public static void CreateSharedDirectory(string path)
```

Comment:
/cc @stephentoub: could you take a look a look at this implementation? This is working around the lack of Linux file permission APIs in .NET.